council/prompt/llm_prompt_config_object.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Mapping, Optional, Sequence
+
+import yaml
+from council.utils import DataObject, DataObjectSpecBase
+
+
+class LLMPromptTemplate:
+    def __init__(self, template: str, model: Optional[str], model_family: Optional[str]) -> None:
+        self._template = template
+        self._model = model
+        self._model_family = model_family
+
+        if self._model is None and self._model_family is None:
+            raise ValueError("At least one of `model` or `model-family` must be defined")
+
```

Comment:
do we want to verify that model and model-family are compliant?
like model.startwidth(model_family) and if not raising a ValueError?